Library/SharedFunctions.swift

```diff
@@ -174,3 +174,19 @@ public func ksr_isOSVersionAvailable(_ version: Double) -> Bool {
 
   return false
 }
+
+public typealias AttributedLinkData = (text: String, url: URL?, attributes: [NSAttributedString.Key: Any])
+
+public func ksr_attributedString(_ attributedString: NSAttributedString,
```

Comment:
Looking at this, I'm seeing some duplication so I'm not sure what was the original decision to make this a function as oppose to an extension function on `NSAttributedString`?

I wonder if we can:

1) Make it an extension on `NSMutableAttributedString` (since we're already passing one in as a param)

```swift
extension NSMutableAttributedString {
  public var /* or func */ copyWith(links: [AttributedLinkData]) -> NSAttributedString {
    // I chose copy because if I remember correctly it usually represented conversion from mutable to an immutable object in ObjC 🤔 
  }
}
```

so then we can call it like so:

```swift
mutableAttributedString.with(links: [(Strings.Learn_more_about_accountability(), url, [:])])
```

2) Maybe make it more readable like this

```swift
let linkData = AttributedLinkData(
  text: Strings.Learn_more_about_accountability(),
  url: url,
  attributes: [:]
)

mutableAttributedString.with(links: [linkData])
```

3) Possibly make attributes optional with default value like so:

```swift
public typealias AttributedLinkData = (
  text: String, 
  url: URL?, 
  attributes: [NSAttributedString.Key: Any]? = [:])
```

so that we can shorten the above to

```swift
let linkData = AttributedLinkData(
  text: Strings.Learn_more_about_accountability(),
  url: url
)
```

Reply:
On second thought this could even be extension on `String` in case we want to assure not mixing styles?

Reply:
You can have defaults in a `typealias`?